imap-proto/src/parser/rfc5464.rs

```diff
@@ -0,0 +1,273 @@
+//!
+//! https://tools.ietf.org/html/rfc5464
+//!
+//! IMAP METADATA extension
+//!
+
+// rustfmt doesn't do a very good job on nom parser invocations.
+#![cfg_attr(rustfmt, rustfmt_skip)]
+#![cfg_attr(feature = "cargo-clippy", allow(redundant_closure))]
+
+
+use nom::IResult;
+
+use std::str;
+use crate::core::*;
+use crate::types::*;
+
+fn is_entry_component_char(c: u8) -> bool {
+    c < 0x80 &&
+        c > 0x19 &&
+        c != b'*' &&
+        c != b'%' &&
+        c != b'/'
+}
+
+enum EntryParseStage<'a> {
+    PrivateShared(usize),
+    Admin(usize),
+    VendorComment(usize),
+    Path(usize),
+    Done(usize),
+    Fail(nom::Err<&'a [u8]>),
+}
+
+fn check_private_shared(i: &[u8]) -> EntryParseStage {
+    if i.starts_with(b"/private") {
+        return EntryParseStage::VendorComment(8);
+    }
+    if i.starts_with(b"/shared") {
+        return EntryParseStage::Admin(7);
+    }
+    return EntryParseStage::Fail(
+        nom::Err::Error(
+        b"Entry Name doesn't start with /private or /shared"));
```

Comment:
Please put the closing parenthesis on the next line here.

Reply:
May be it is better to put error message on the same line as  "...Error("

Reply:
Please do whatever `rustfmt` does in this case.